### ISSUE TITLE
[G4.0.6] Inexplicable `Paused on XMLHttpRequest`

### ISSUE THREAD

**Issue by DonielMoins** [BrowserWorks/waterfox#2539]
### What happened?

#A bug happened!

A “Paused on XMLHttpRequest” breakpoint triggers with no breakpoints set. It is… very annoying when I have to work with web sockets on a daily-basis. 

So this happens when I open the “DevTools” menu and open either the debugger or the Network tab.
I used my same FF sync account on another Waterfox instance (same version) on a Windows machine, out of desperation, and it wasn't happening again.

This issue is NOT site specific, and I have disabled all relevant breakpoint options from every option menu I can find. 
I made sure to test if this still persists with all extensions disabled  (pretty sure this was a while ago).

Furthur testing shows that this problem does not still occur in Private Browsing mode.

I will attach any relevant screenshots and logs I can think of. If you need any other information, please don't hesitate to ask.

---

[![https://imgur.com/h6fbX4y.png](https://imgur.com/h6fbX4y.png)](https://imgur.com/h6fbX4y.png)

`this.content = {
    XMLHttpRequest: window.XMLHttpRequest,
    fetch: window.fetch.bind(window),
};
window.JSON = JSON;
window.XMLHttpRequest = XMLHttpRequest;
window.fetch = fetch;`

Funnily enough, I found this suspicous ExtensionContent.jsm under resource://gre/modules (shown in the image below)
[![https://imgur.com/GYH6Bh8.png](https://imgur.com/GYH6Bh8.png)](https://imgur.com/GYH6Bh8.png)
but I doubt it has any relation since it is also present in PB mode.

[![https://imgur.com/GCaqu5X.png](https://imgur.com/GCaqu5X.png)](https://imgur.com/GCaqu5X.png)
Here is where it breaks in the jQuery JS file. Seems like a normal XHR request right?

---

##XHR related settings here
[![https://imgur.com/aoEfyj3.png](https://imgur.com/aoEfyj3.png)](https://imgur.com/aoEfyj3.png)
[![https://imgur.com/l47ER75.png](https://imgur.com/l47ER75.png)](https://imgur.com/l47ER75.png)
[![https://imgur.com/hEFI4Gn.png](https://imgur.com/hEFI4Gn.png)](https://imgur.com/hEFI4Gn.png)

##Other settings:
[![https://imgur.com/kAFLdkW.png](https://imgur.com/kAFLdkW.png)](https://imgur.com/kAFLdkW.png)
[![https://imgur.com/RUEwU3Q.png](https://imgur.com/RUEwU3Q.png)](https://imgur.com/RUEwU3Q.png)

I'm fairly certain *I* caused this but forgot what setting I changed after a long break from web development.
But at this point... I am despairing. I cannot find how to fix this.

Thank you all,
- F H

### Reproducible?

- [x] I have tested this on Firefox.

### Version

G4.𝑋 (Future)

### What platform are you seeing the problem on?

Linux

### Relevant log output

`23:34:31.390 Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 2 background.js:2807
23:34:31.597 Resource  of root-node was not found. resource-command.js:724:17
23:34:31.646 Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 2 background.js:2807
23:34:31.926 Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 2 background.js:2807
23:34:32.910 Use of nsIFile in content process is deprecated. 2 NetUtil.jsm:253:8
23:34:32.951 Use of nsIFile in content process is deprecated. NetUtil.jsm:253:8
23:34:33.099 Error: Can't find profile directory. 2 XULStore.jsm:66:15
23:34:33.351 Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 2 background.js:2807
23:34:33.950 Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 2 background.js:2807
23:34:52.794 Type error for parameter tabId (Integer -1 is too small (must be at least 0)) for tabs.get. background.js:511`

**Comment by DonielMoins** [comment]
Adding upon the "Can't find profile directory" error.
`23:34:33.099 Error: Can't find profile directory. 2 XULStore.jsm:66:15
    load resource://gre/modules/XULStore.jsm:66
    XULStore resource://gre/modules/XULStore.jsm:24
    enter resource://devtools/server/actors/utils/event-loop.js:79
    enter self-hosted:1175
    _pushThreadPause resource://devtools/server/actors/thread.js:334
    _pauseAndRespond resource://devtools/server/actors/thread.js:956
    _onOpeningRequest resource://devtools/server/actors/thread.js:782
    _onOpeningRequest self-hosted:1175
    jQuery 2
    Ajax http://server01.evilcorp.wg/js/rtorrent.js:1167
    requestWithoutTimeout http://server01.evilcorp.wg/js/webui.js:2615
    getUISettings http://server01.evilcorp.wg/js/webui.js:331
    success http://server01.evilcorp.wg/js/rtorrent.js:1232
    jQuery 6
    Ajax http://server01.evilcorp.wg/js/rtorrent.js:1167
    requestWithoutTimeout http://server01.evilcorp.wg/js/webui.js:2615
    getPlugins http://server01.evilcorp.wg/js/webui.js:314
    init http://server01.evilcorp.wg/js/webui.js:222
    <anonymous> http://server01.evilcorp.wg/js/webui.js:2672
    jQuery 8
`

**Comment by adamp01** [comment]
@DonielMoins - So with a fresh profile this issue does not occur?

**Comment by DonielMoins** [comment]
nope. but that still begs the question, how did I mess this up so bad that even after days of rummaging through settings, I couldn't fix it without a 'hard reset'. I wish I could have backed up my corrupted waterfox...

**Comment by DonielMoins** [comment]
Sorry I'm about to head to sleep rn, but I think I did back it up just to fix it properly later on. So maybe we can help so.eone out later on? I hope you don't mind this, I think it would be extremely beneficial. Might even be able to fix a bug or improve a broken part of Firefox? Or serve as a guide at the very least.

**Comment by adamp01** [comment]
If you can send me a list of modified preferences in your broken profile I'm happy to have a look through and see if anything stands out.

**Comment by DonielMoins** [comment]
Weirdly, the only files related I can find are .js suffixed. Are the prefs*.js files the ones you're talking about? 

> If you can send me a list of modified preferences in your broken profile I'm happy to have a look through and see if anything stands out.

**Comment by adamp01** [comment]
@DonielMoins - apologies for the really late reply here - yeah the contents of `prefs.js` in your main profile would be what I'm after here.

**Comment by allan-bonadio** [comment]
I've got a similar thing happening.  On Windows.  And, a similar request I'm developing at this time, doesn't stop there.  v114 firefox.

OOps! my mistake.  'XHR Breakpoints' > "Pause on any URL" was on.  Maybe that's the problem.
